### DISCUSSION TITLE
eval_int in Python API

### DISCUSSION THREAD

**Discussion by smithnormform** [GaloisInc/saw-script#1641] [Q&A]
Is there a Python API function similar to the SAWScript `eval_int`?

For example, say I have a Cryptol specification that's the instantiation of a parameterized module with a parameter `size` where `size` is the size of an array. In SAWScript I would write something like:

```
let someSize = eval_int {{ `size : [64] }};
(x, xp) <- ptr_to_fresh "x" (llvm_array someSize i32);
```

**Comment by chameco** [comment]
I don't think there is currently an equivalent to `eval_int` in Cryptol - there probably should be though!

(Somewhat-relatedly, there's a cool SAWScript function `eval_size` that directly takes a Cryptol size type and produces an `Int`, which makes this common case a bit less verbose. That also isn't currently accessible from Python, and also probably should be.)

**Reply by chameco** [reply]
Tracking this in #1642

**Reply by smithnormform** [reply]
Is there any update on this? My work around for now has been to either hard code in values (less than ideal if new instances of the parameterized module are made) or pass in values as arguments when initializing contracts (also less than ideal).

**Reply by chameco** [reply]
I've implemented something that might help in #1660.
